overlord/state/taskrunner.go

```diff
@@ -63,6 +63,21 @@ func (r *TaskRunner) Handlers() map[string]HandlerFunc {
 	return r.handlers
 }
 
+// taskFail marks task t and all tasks waiting (directly and indirectly on it) as in ErrorStatus.
+func taskFail(t *Task) {
+	t.SetStatus(ErrorStatus)
+	mark := append([]*Task(nil), t.HaltTasks()...)
+	i := 0
+	for i < len(mark) {
+		t := mark[i]
+		if t.Status() == WaitingStatus {
+			t.SetStatus(ErrorStatus)
+			mark = append(mark, t.HaltTasks()...)
```

Comment:
this is done all with the state lock held, also t is not the same in the two places, I should maybe use two variable names